core/src/main/scala/variant/Antichess.scala

```diff
@@ -103,3 +98,9 @@ case object Antichess
     // The pawn cannot attack a bishop or be attacked by a bishop
     val cannotAttackBishop = pawn.isLight != oppositeBishopLight
     InsufficientMatingMaterial.pawnBlockedByPawn(pawn, position) && cannotAttackBishop
+
+  private def allOnSameColourSquares(position: Position): Boolean =
+    position.white.squares.forall(_.isLight) == position.black.squares.forall(_.isLight)
+
+  private def justOneKnightEach(position: Position): Boolean =
+    position.onlyKnights && position.white.squares.size == 1 && position.black.squares.size == 1
```

Comment:
instead of requesting the [list of squares](https://github.com/lichess-org/scalachess/blob/player-insufficient-material/core/src/main/scala/Bitboard.scala#L110-L110) then counting the list, you could do a much faster [bit count](https://github.com/lichess-org/scalachess/blob/player-insufficient-material/core/src/main/scala/Bitboard.scala#L76-L77)